slash-commands/praise/index.ts

```diff
@@ -51,14 +59,23 @@ const praise = async (interaction) => {
     }
 };
 
-const getPraiseText = (interaction, mention, shouldShame) => {
+const getPraiseText = (
+    interaction: CommandInteraction,
+    mention: GuildMember | Role | APIRole | User,
+    shouldShame: boolean
+) => {
     const praiser = getNicknameOrName(interaction);
     const actionType = shouldShame ? "shames" : "praises";
     let messageText = `${praiser} ${actionType} `;
-    if (mention.user) {
-        messageText += `${createUserMentionWithId(mention.user.id)}!`;
+    if ((<GuildMember>mention).user) {
+        messageText += `${createUserMentionWithId(
+            (<GuildMember>mention).user.id
+        )}!`;
     } else {
-        if (mention.name === "@everyone") {
+        if (
+            (<Role | APIRole>mention).name &&
+            (<Role | APIRole>mention).name === "@everyone"
+        ) {
```

Comment:
tähän vois tehä type guardit / type predicatet, parempi käytäntö ku type castaaminen, ja niitä vois käyttää jatkossa myös vastaavissa tilanteissa. https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates

Reply:
Menis about näin:
```
export const isRole = (mention: GuildMember | Role | APIRole | User): mention is Role | APIRole => {
    return (mention as Role).user != null
}
```